src/cli/rustup_mode.rs

```diff
@@ -1312,10 +1320,27 @@ fn toolchain_link(cfg: &Cfg, m: &ArgMatches<'_>) -> Result<utils::ExitCode> {
 }
 
 fn toolchain_remove(cfg: &mut Cfg, m: &ArgMatches<'_>) -> Result<utils::ExitCode> {
-    for toolchain in m.values_of("toolchain").unwrap() {
-        let toolchain = cfg.get_toolchain(toolchain, false)?;
-        toolchain.remove()?;
+    if m.is_present("pattern") {
+        for pattern_str in m.values_of("toolchain").unwrap() {
+            let pattern = Pattern::new(&pattern_str)?;
+
+            let mut toolchains = cfg.get_toolchains_from_glob(pattern)?.peekable();
+            if toolchains.peek().is_none() {
+                info!("no toolchains matched pattern '{}'", pattern_str);
+                return Ok(utils::ExitCode(0));
```

Comment:
I think this should return a non-zero exit code, but I modelled this after `rustup toolchain uninstall some-toolchain`, which just shows a message with a successful exit code. Should I change both of them to use exit code 1?

Reply:
If it has always been "safe" in the shell scripting sense to "remove" a toolchain which isn't present, then we wouldn't want to break peoples' scripting by making them check first before trying.  Better to be informational but consistent with past behaviour here I think.

Reply:
I'm not entirely sure that this makes sense. Patterns are new; failure to match a pattern is a new situation. We can detect a toolchain string that has no pattern characters and do the prior behaviour, if we want to.

That means, we can take a first principles design here if appropriate.